### ISSUE TITLE
False negative : overriding trait method visibility of another trait

### ISSUE THREAD

**Issue by Kallys** [vimeo/psalm#6025]
Inception of #1175 returns false negative.

```php
<?php

trait T1 {
  protected function f(): void {}
}

trait T2 {
  use T1 {
    f as public;
  }
}

class C {
  use T2;
}

(new C)->f();
```

Expected: no issue
Actual: InaccessibleMethod - https://psalm.dev/r/e4f107a187
